### DISCUSSION TITLE
More features for show_in_material_panel button for node properties

### DISCUSSION THREAD

**Discussion by Psi25** [bnpr/Malt#397] [Feedback & Feature Requests]
It's nice having control over node properties by showing what can be edited in the material panel, but it should have more features. 

Simply ticking off the eye to show it in material panel then ticking off the checkbox of the property to edit the variables locally isn't efficient enough. It would be helpful if there were an option to automatically show a node property  while ticked as locally editable.

![image](https://user-images.githubusercontent.com/103315144/191907854-6f86bbf2-9572-46c5-8344-32588270e54d.png)

**Comment by pragma37** [comment]
So, do you mean enabling the override by default when the parameter is shown in the material?

**Reply by Psi25** [reply]
Yes.

**Reply by pragma37** [reply]
That would mean that once a parameter is exposed you can't change the default so it propagates to all the materials that use it.  
So I don't think is a good idea at the moment.  
It should be possible to propagate the changes when the material value matches the default, but the current system can't support that.

**Reply by Psi25** [reply]
It would be nice to have as a feature. It's nice that you can show certain settings in the material, but having to tick the box so you can change a setting of a certain material is a slowdown.

**Reply by HSDSZ** [reply]
> It would be nice to have as a feature. It's nice that you can show certain settings in the material, but having to tick the box so you can change a setting of a certain material is a slowdown.

I don't really agree with you on this feature. For example, I have several different materials using the same malt nodes. If override is enabled by default, then I cannot change their outline width (for example) at the same time

**Reply by Psi25** [reply]
> If override is enabled by default, then I cannot change their outline width (for example) at the same time.

That makes sense, but what if there are 2 buttons instead? One for local override and another for global override. It's a simple solution but it could work.
